src/pages/experiments/[experimentId]/data-processing/data-integration/components/CalculationConfig.jsx

```diff
@@ -55,7 +56,44 @@ const CalculationConfig = () => {
     },
   ];
 
-  const changesOutstanding = false;
+  const initialState = {
+    dataIntegration: {
+      method: 'seuratv4',
+      noGenes: 2000,
+      normalisation: 'logNormalise',
+    },
+    dimensionalityReduction: {
+      numPCs: 30,
+      variationExplained: 91,
+      excludeGeneCategories: [],
+      method: 'rpca',
+    },
+  };
+
+  const [changesOutstanding, setChangesOutstanding] = useState(false);
+  const [settings, setSettings] = useState(initialState);
+  const [savedSettings, setSavedSettings] = useState(initialState);
+
+  const updateSettings = (diff) => {
+    const newSettings = _.cloneDeep(settings);
+
+    // Customizer function to replace value of array in object
+    const arrayMerge = (obj, src) => {
+      if (_.isArray(obj)) {
+        return src;
+      }
+    };
+
+    _.mergeWith(newSettings, diff, arrayMerge);
+
+    if (!_.isEqual(newSettings, savedSettings)) {
```

Comment:
rather than having one state update triggering a different state update, perhaps we could set changesOutstanding as a const = !_.isEqual(settings, savedSettings)?

Reply:
you mean :
```
  setSettings(newSettings);
  setChangesOutstanding(!_.isEqual(settings, savedSettings))
```
like that?

Reply:
no, I mean setting the comparison in <alert

Reply:
I don't know what is a better practice for react though, calculating it on each render or adding an extra usestate

Reply:
Sorry, I don't understand 😅 I think the current implementation is ok already, comparing the previous state to the current state on every state change. Of course, the state is not yet updated, that's why the alert will disappear if the user returns to the settings used on the last run

Reply:
I meant having `const changesOutstanding = !_.isEqual(settings, savedSettings)`, not a useState.

Reply:
We need `changesOutstanding` to be recomputed every time there is a settings change. It's implemented as a React states, so we can use `setChangeOutstanding` to update the data on event change. Writing it as `const changesOutstanding = !_.isEqual(settings, savedSettings)` does not set it up as a React state, so it's not recalculated.  If we want to write it like that, I think we'd have to put it in a `useEffect(() => {}, [settings])` block, so that it'll be recomputed everytime `settings` change.